model/testing/src/icon4py/model/testing/grid_utils.py

```diff
@@ -134,7 +134,9 @@ def get_grid_geometry(
     on_gpu = data_alloc.is_cupy_device(backend)
     xp = data_alloc.array_ns(on_gpu)
     num_levels = get_num_levels(experiment)
-    register_name = experiment.join(backend.name)
+    register_name = experiment.join(
+        backend.name if not helpers.is_embedded(backend) else str(backend)
```

Comment:
For embedded `backend==None`, then the type annotation for `backend` needs to be updated to e.g. `gtx_backend.Backend | None`.

Reply:
Maybe it would make sense to introduce a `backend_name` function, as this line of code is already duplicated and might be needed in more places.